src/test/java/io/vertx/core/spi/executor/Utils.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.spi.executor;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.concurrent.ExecutorService;
+import java.util.logging.Level;
+
+import org.junit.rules.ExternalResource;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.test.core.LogContainsRule;
+
+public class Utils {
+
+  private static final Logger log = LoggerFactory.getLogger(ExecutorServiceFactory.class);
+
+  static final String SERVICES_SRC_DIR = "src/test/classpath/servicehelper/META-INF/services/";
+
+  static final String TEST_CLASSPATH = "target/test-classes/";
+  static final String SERVICE_LOADER_SUFFIX = "META-INF/services/";
+  static final String SERVICE = "io.vertx.core.spi.executor.ExecutorServiceFactory";
+  static final String SERVICE_CFG_FILE = TEST_CLASSPATH + SERVICE_LOADER_SUFFIX + SERVICE;
+
+  static final String TESTS_PACKAGE = Utils.class.getPackage().getName(); // TODO replace with clazz.getPackage()
+
+
+  /*
+   * Set up some typesafe tokens for the testcases to refer to ExecutorService
+   * methods
+   */
+  static Method EXECUTE = null;
+  static Method SHUTDOWN_NOW = null;
+  static {
+    try {
+      EXECUTE = ExecutorService.class.getMethod("execute", Runnable.class);
+      SHUTDOWN_NOW = ExecutorService.class.getMethod("shutdownNow");
+    } catch (NoSuchMethodException | SecurityException e) {
+      throw new IllegalStateException(e);
```

Comment:
I think that will prevent the class loading instead we should set null fields and fail in the other methods with an assertion that check the fields are not null before using them or throwing an IllegalStateException at this place

Reply:
Ok I will make it so. Originally the methods parameters were not typed, only string names but what you suggest is better now they are the actual method types.

Reply:
Done, is ready for another review @vietj I think. Thank you again.